### ISSUE TITLE
Build-ScopeTableForDeploymentRootScope.ps1 tries to enumerate subscriptions in different tenant via Lighthouse

### ISSUE THREAD

**Issue by artisticcheese** [Azure/enterprise-azure-policy-as-code#546]
**Describe the bug**
Newer version (10x) seems to be trying to get access to management groups for subscriptions which are not in current tenant (via Lighthouse) and fails as a result. Example is below where my account is having access to some resources in external tenant via Lighthouse

**To Reproduce**
Onboard external resource via Lighthouse into tenant where EPAC shall be working on

**Expected behavior**
This behavior shall not exist at all since policies are being applied only to current tenant

**Screenshots**
![image](https://github.com/Azure/enterprise-azure-policy-as-code/assets/13176884/0e189991-e946-4669-ac07-5d64047e5d4e)

**EPAC Version**
10.0.2

**Comment by gregslack78** [comment]
@artisticcheese - Can you provide the failure output from the pipeline or debugger?
Are you running EPAC against the lighthouse (managed) tenant or the "parent" tenant?

**Comment by artisticcheese** [comment]
Here is exact error. This is running under service principal (5b3a2e7a-XXXX-4cec-957a-43e2826d5de0) in Tenant 1 which is through Lighthouse have access to subscription (43909631-XXXX-4b09-9fdb-706fc002ae5f) in Tenant2.

```
Loaded Module 'Az.Accounts'
Loaded Module 'Az.MSGraph'
Get-AzManagementGroup: /home/vsts/work/1/s/Scripts/Helpers/Build-ScopeTableForDeploymentRootScope.ps1:106
Line |
 106 |  … mentGroup = Get-AzManagementGroup -GroupName $deploymentRootScopeMana …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | The client '5b3a2e7a-XXXX-4cec-957a-43e2826d5de0' with object id
     | '5b3a2e7a-XXXX-4cec-957a-43e2826d5de0' does not have authorization to
     | perform action 'Microsoft.Management/register/action' over scope
     | '/subscriptions/43909631-XXXX-4b09-9fdb-706fc002ae5f' or the scope is
     | invalid. If access was recently granted, please refresh your credentials.

```

**Comment by artisticcheese** [comment]
Add this subscription to exclusion does not help either
```
 {
         "pacSelector": "prod-environment",
         "cloud": "AzureCloud",
         "tenantId": "d7585f83-XXXX-492f-9b2f-34717df7700c",
         "keepDfcSecurityAssignments": true,
         "deploymentRootScope": "/providers/Microsoft.Management/managementGroups/DevOps-mg", // replaces rootScope
         "globalNotScopes": [
            "/providers/Microsoft.Management/managementGroups/security-mg",
            "/subscriptions/43909631-XXXX-4b09-9fdb-706fc002ae5f"
         ],
         "desiredState": { 
            "keepDfcSecurityAssignments": true,
            "strategy": "ownedOnly", 
            "excludedScopes": [
               "/providers/Microsoft.Management/managementGroups/security-mg"
            ],
            "excludedPolicyDefinitions": [
            ],
            "excludedPolicySetDefinitions": [
            ],
            "excludedPolicyAssignments": [
            ]
         },
         "managedIdentityLocation": "eastus2"
      }
   ]
```

**Comment by artisticcheese** [comment]
This seems to be caused what is default context is chosen for `Connect-AzAccount` cmdlet. If default subscription is not specified and subscription in Lighthouse tenant is chosen then  `Get-AzManagementGroup` would fail. You can see reproduction scenario below. Default subscription in this case after `Login-AzAccount` is executed and set to subscription in Lighthouse tenant and first call to `Get-AzManagementGroup` cmdlet fails, if I switch AzContext to subscription in home tenant then issue is not present.

```
PS C:\repo\azure-policies-infra> Get-AzSubscription

Name                              Id                                   TenantId                             State
----                              --                                   --------                             -----
InfoSec Subscription - Production 43909631-XXXX-4b09-9fdb-706fc002ae5f d7585f83-XXXX-492f-9b2f-34717df7700c Enabled
Dev controlplane 1                45b6efbb-XXXX-4a2b-915c-fb954716b0d2 d7585f83-XXXX-492f-9b2f-34717df7700c Enabled
Control Plane 1                   0973ccdf-XXXX-4438-8509-67c1f84ae5a5 d7585f83-XXXX-492f-9b2f-34717df7700c Enabled
Prod 1                            9893b814-XXXX-4d2e-a2aa-cc8d383953ec d7585f83-XXXX-492f-9b2f-34717df7700c Enabled
Sandbox 1                         820b3d00-XXXX-4adf-b34b-a3fe8ca88aca d7585f83-XXXX-492f-9b2f-34717df7700c Enabled

PS C:\repo\azure-policies-infra> Get-AzManagementGroup -GroupName "DevOps-mg" -Expand -Recurse          
Get-AzManagementGroup: The client '5b3a2e7a-XXXX-4cec-957a-43e2826d5de0' with object id '5b3a2e7a-XXXX-4cec-957a-43e2826d5de0' does not have authorization to perform action 'Microsoft.Management/register/action' over scope '/subscriptions/43909631-XXXX-4b09-9fdb-706fc002ae5f' or the scope is invalid. If access was recently granted, please refresh your credentials.
PS C:\repo\azure-policies-infra> Get-AzContext

Name                                     Account                                             SubscriptionName                                   Environment                                        TenantId
----                                     -------                                             ----------------                                   -----------                                        --------
InfoSec Subscription - Production (4390… 4ec14295-XXXX-4a8c-95d4-3abc6cf33776                InfoSec Subscription - Production                  AzureCloud                                         d7585f83-XXXX-492f-9b2f-34717df7700c

PS C:\repo\azure-policies-infra> Set-AzContext -Subscription "Dev controlplane 1"

Name                                     Account                                             SubscriptionName                                   Environment                                        TenantId
----                                     -------                                             ----------------                                   -----------                                        --------
Dev controlplane 1 (45b6efbb-2c18-4a2b-… 4ec14295-XXXX-4a8c-95d4-3abc6cf33776                Dev controlplane 1                                 AzureCloud                                         d7585f83-XXXX-492f-9b2f-34717df7700c

PS C:\repo\azure-policies-infra> Get-AzManagementGroup -GroupName "DevOps-mg" -Expand -Recurse

Id                : /providers/Microsoft.Management/managementGroups/DevOps-mg
Type              : Microsoft.Management/managementGroups
Name              : DevOps-mg
TenantId          : d7585f83-XXXX-492f-9b2f-34717df7700c
DisplayName       : DevOps Management Group
UpdatedTime       : 7/28/2022 3:30:32 PM
UpdatedBy         : bd5cfa79-0cca-4045-9265-564bbaee64f4
ParentId          : /providers/Microsoft.Management/managementGroups/d7585f83-XXXX-492f-9b2f-34717df7700c
ParentName        : d7585f83-XXXX-492f-9b2f-34717df7700c
ParentDisplayName : Tenant Root Group
Children          : Live Management Group
```

**Comment by artisticcheese** [comment]
Parent of bottom 3

**Comment by gregslack78** [comment]
OK, I think the fix you put in just happens to work, but in many situations may not be the case.  I think if you were to directly add a new subscription under the "Security Management Group" MG in the not-lighthouse tenant, and you were to set the context to that subscription, you would get the same error.  I don't think (Still trying to see if I can find a way to verify this) it has to do with the fact that the subscription set in your context has a different "HomeTenantId", I think it has to do with the fact that it is under the "Security Management Group" MG and you have less access (e.g. don't have 'Microsoft.Management/register/action') at that scope, which going off the image appears to be the case; Resource Access vs. Account Admin

**Comment by gregslack78** [comment]
Can you send the json for the resource access role?

![image](https://github.com/Azure/enterprise-azure-policy-as-code/assets/85254974/f44e112c-66de-43cd-8c4d-01aa29dbf7e1)


![image](https://github.com/Azure/enterprise-azure-policy-as-code/assets/85254974/07136082-20cf-4d50-971a-6bc618cb6a97)

**Comment by artisticcheese** [comment]
I'm confused, you want me to send JSON of built-in `Reader` role?

**Comment by gregslack78** [comment]
No, the resource access role

![image](https://github.com/Azure/enterprise-azure-policy-as-code/assets/85254974/62ddadac-a3ff-4d7a-9eba-135578dbb2c3)

**Comment by artisticcheese** [comment]
I do not have access to that subscription at all, only to some resources in that subscription
![image](https://github.com/Azure/enterprise-azure-policy-as-code/assets/13176884/302cbfb7-e337-41ec-9a12-8dcefbdda251)

**Comment by gregslack78** [comment]
Can you...
1. Go to the Security Management Group
2. Access Control
3. Roles tab
4. Search for "resource access " role
5. View details
6. json tab

**Comment by gregslack78** [comment]
Can you email me at gregslack@microsoft.com to set up some time for a call?

**Comment by gregslack78** [comment]
@artisticcheese - Can you try testing this proposed fix in your environment?
[fix546.zip](https://github.com/Azure/enterprise-azure-policy-as-code/files/15011901/fix546.zip)

**Comment by artisticcheese** [comment]
Yes, this works

**Comment by techlake** [comment]
Fixed in v10.1.0
